src/dsync/dsync.c

```diff
@@ -3186,9 +3189,12 @@ int main(int argc, char **argv)
         }
         usage = 1;
     }
-
+    
+    if (usage) {
+        goto do_usage;
```

Comment:
How about if you just move the code from lines 3225-3229 (if rank==0 print_usage(), goto dsync_common_cleanup) here?  Then you don't need the label `do_usage` and don't need two separate `if (usage) {` blocks, I think.

Reply:
there is still the part with 2 arguments left to check if the source and destination are there. Can't move it

Reply:
I moved a litle more so it yields the parser failures and the error cause we don't have source and destination argument if they're missing . that's even bettoer.